ADB_world/Worldmap_year/OneYearMap.py

```diff
@@ -0,0 +1,163 @@
+                  "07-July", "08-August", "09-September", "10-October", "11-November", "12-December"]
+
+    # Generate flight connection data for each month and save it
+    for month in month_list:
+        def_flight_connection.generate_flight_connections_json(month)
+
+    # Process flight connections for January to create initial DataFrame
+    connections_df, _ = def_flight_connection.process_flight_connections("01-January")
+
+    # Iterate over month_list to gather and concatenate data
+    for month in month_list[1:]:
+        connections_data, _ = def_flight_connection.process_flight_connections(month)
+        connections_df = pd.concat([connections_df, connections_data], ignore_index=True)
+
+    # Consolidate connections with the same departure and destination
+    connections_df = connections_df.groupby(["icao_departure", "icao_destination"], as_index=False).agg({
+        "departure_airport_name": "first",
+        "lat_departure": "first",
+        "lon_departure": "first",
+        "lat_destination": "first",
+        "lon_destination": "first",
+        "averageDailyFlights": lambda x: x.sum() / 12  # Divide the sum by 12 for annual average
+    })
+
+    # Convert DataFrame to dictionary and save as JSON file
+    all_connections.append(connections_df.to_dict(orient='records'))
+    with open(file_path, 'w') as f:
+        json.dump(all_connections, f, indent=4)
+    print(f"All connections saved to {file_path}")
+else:
+    print(f"The file '{file_path}' already exists.")
+
+#######################################
+# Plotting ############################
+#######################################
+
+# Process flight connection data for the entire year
+flight_data_df, daily_flights_df = def_flight_connection.process_flight_connections("Year")
+
+# Calculate the maximum value of 'daily_flights' column
+max_daily_flights = daily_flights_df['number_of_total_flights'].max()
+
+# number of total connections
+total_connections = len(flight_data_df)
+
+# Define maximum and minimum sizes for markers
+max_size = 20
+min_size = 2
+
+# scale the values between min and max sizes
+scaled_sizes = ((daily_flights_df['number_of_total_flights'] - daily_flights_df['number_of_total_flights'].min()) /
+                (daily_flights_df['number_of_total_flights'].max() - daily_flights_df['number_of_total_flights'].min()) *
+                (max_size - min_size)) + min_size
+
+# Plot the different airport markers
+fig = go.Figure(go.Scattergeo(
+    lat=daily_flights_df["lat_departure"],
+    lon=daily_flights_df["lon_departure"],
+    text=daily_flights_df["departure_airport_name"] + '<br>'
+    + "Number of average departing daily flights: "
+    + daily_flights_df["number_of_total_flights"].apply(lambda x: round(x, 2)).astype(str),  # Round to 2 decimal places
+    hoverinfo='text',  # Set hoverinfo to 'text'
+    mode='markers',
+    marker=dict(
+        size=scaled_sizes,  # Scale the sizes between min and max sizes
```

Comment:
Can I get a legend for that also? Currently, there is only a colormap legend.

Reply:
I don't know if a second legend would make sense since the scaling for color and size is the same?

Reply:
@michaelweinold